core/src/main/java/psiprobe/controllers/deploy/CopySingleFileController.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed under the GPL License. You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+ * 
+ * THIS PACKAGE IS PROVIDED "AS IS" AND WITHOUT ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,
+ * WITHOUT LIMITATION, THE IMPLIED WARRANTIES OF MERCHANTIBILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE.
+ */
+
+package psiprobe.controllers.deploy;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Vector;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.catalina.Context;
+import org.apache.commons.fileupload.FileItem;
+import org.apache.commons.fileupload.FileItemFactory;
+import org.apache.commons.fileupload.FileUploadBase;
+import org.apache.commons.fileupload.disk.DiskFileItemFactory;
+import org.apache.commons.fileupload.servlet.ServletFileUpload;
+import org.apache.commons.fileupload.servlet.ServletRequestContext;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.view.InternalResourceView;
+
+import psiprobe.controllers.TomcatContainerController;
+
+/**
+ * Uploads and installs web application from a .WAR.
+ * 
+ * @author Vlad Ilyushchenko
+ * @author Mark Lewis
+ */
+public class CopySingleFileController extends TomcatContainerController {
+
+  /** The Constant logger. */
+  private static final Logger logger = LoggerFactory.getLogger(CopySingleFileController.class);
+
+  @Override
+  protected ModelAndView handleRequestInternal(HttpServletRequest request,
+      HttpServletResponse response) throws Exception {
+
+	  List<Context> apps;
+	    try {
+	      apps = getContainerWrapper().getTomcatContainer().findContexts();
+	    } catch (NullPointerException ex) {
+	      throw new IllegalStateException("No container found for your server: "
+	          + getServletContext().getServerInfo(), ex);
+	    }
+	  
+	    Vector applications = new Vector();
```

Comment:
Can you use something other than vector here?  Vector is a bad idea for new code.